src/main/java/project/OlafApp.java

```diff
@@ -0,0 +1,174 @@
+package project;
+
+import project.common.PrintText;
+import project.io.Parser;
+import project.io.Ui;
+import project.storage.Storage;
+import project.task.Deadline;
+import project.task.Event;
+import project.task.Task;
+import project.task.TaskList;
+import project.task.Todo;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.LocalDateTime;
+
+/**
+ * Handles application logic.
+ */
+public class OlafApp {
+    private TaskList taskList;
+    private Ui ui;
+    private Storage storage;
+    private boolean isActive;
+
+    /**
+     * Creates an instance of {@code OlafApp}.
+     *
+     * @param tasks All the tasks stored.
+     * @param ui A {@code Ui} instance to handle test output.
+     * @param storage A {@code Storage} instance to handle saving and loading tasks from local file.
+     */
+    OlafApp(TaskList tasks, Ui ui, Storage storage) {
+        this.taskList = tasks;
+        this.ui = ui;
+        this.storage = storage;
+        this.isActive = true;
+    }
+
+    /**
+     * Executes each session of the application.
+     */
+    public void run() {
```

Comment:
This method seems a little long. Would it be better to abstract out some of the logic into other methods instead?

Reply:
I'm thinking about it too! I've got some ideas but will have to refactor things and see how it goes :)